src/app/client/src/app/modules/dashboard/components/list-all-reports/list-all-reports.component.ts

```diff
@@ -0,0 +1,147 @@
+import { ResourceService } from '@sunbird/shared';
+import { Component, OnInit } from '@angular/core';
+import { catchError, mergeMap, map, tap } from 'rxjs/operators';
+import * as _ from 'lodash-es';
+import { ReportService } from '../../services';
+import { of, Observable, throwError } from 'rxjs';
+import { ActivatedRoute, Router } from '@angular/router';
+import * as moment from 'moment';
+
+@Component({
+  selector: 'app-list-all-reports',
+  templateUrl: './list-all-reports.component.html',
+  styleUrls: ['./list-all-reports.component.scss']
+})
+export class ListAllReportsComponent implements OnInit {
+
+  constructor(public resourceService: ResourceService, private reportService: ReportService, private activatedRoute: ActivatedRoute,
+    private router: Router) { }
+
+  public reportsList$: Observable<any>;
+  public noResultFoundError: string;
+  private _isUserReportAdmin: boolean;
+
+  ngOnInit() {
+    this.reportsList$ = this.reportService.isAuthenticated(_.get(this.activatedRoute, 'snapshot.data.roles')).pipe(
```

Comment:
yeah, that's true. 
But I've done it on purpose because if a user is not authenticated then there is no need of checking the roles whether he' admin or not.  We're directly throwing error at this time. With forkJoin or combineLatest it subscribes to both the observables which are not required.

If he's authenticated then only it should be checked whether he is admin or not and need to call the api accordingly.